internal/backends/winit/winitwindowadapter/accesskit.rs

```diff
@@ -0,0 +1,437 @@
+        self.inner.update_if_active(|| TreeUpdate {
+            nodes: vec![],
+            tree: None,
+            focus: new.map(|item| node_id_for_item_rc(item, self.tree_generation.get())),
+        })
+    }
+
+    fn handle_request(&self, request: ActionRequest) {
+        let Some(window_adapter) = self.window_adapter_weak.upgrade() else { return };
+        match request.action {
+            Action::Focus => {
+                if let Some(item) = item_rc_for_node_id(request.target, self.tree_generation.get())
+                {
+                    WindowInner::from_pub(window_adapter.window()).set_focus_item(&item);
+                }
+            }
+            _ => {}
+        }
+    }
+
+    pub fn unregister_component<'a>(&self, _component: ComponentRef) {
+        self.all_nodes.borrow_mut().clear();
+
+        self.tree_generation.set(self.tree_generation.get() + 1);
+
+        let win = self.window_adapter_weak.clone();
+        i_slint_core::timers::Timer::single_shot(Default::default(), move || {
+            if let Some(window_adapter) = win.upgrade() {
+                let self_ = &window_adapter.accesskit_adapter;
+                self_.inner.update_if_active(|| self_.build_new_tree())
+            };
+        });
+    }
+
+    fn rebuild_tree_of_dirty_nodes(&self) {
+        if !self.global_property_tracker.is_dirty() {
+            return;
+        }
+
+        // It's possible that we may have been triggered by a timer, but in the meantime
+        // the node tree has been emptied due to a tree structure change.
+        if self.all_nodes.borrow().is_empty() {
+            return;
+        }
+
+        let Some(window_adapter) = self.window_adapter_weak.upgrade() else { return };
+        let window = window_adapter.window();
+
+        self.inner.update_if_active(|| {
+            self.global_property_tracker.as_ref().evaluate_as_dependency_root(|| {
+                let scale_factor = ScaleFactor::new(window.scale_factor());
+
+                let all_nodes = self.all_nodes.borrow();
+
+                let nodes = all_nodes.iter().filter_map(|cached_node| {
+                    if !cached_node.tracker.is_dirty() {
+                        return None;
+                    }
+                    cached_node
+                        .tracker
+                        .as_ref()
+                        .evaluate_if_dirty(|| {
+                            let item =
+                                item_rc_for_node_id(cached_node.id, self.tree_generation.get())?;
+
+                            let mut builder = self.build_node_without_children(&item, scale_factor);
+
+                            builder.set_children(cached_node.children.clone());
+
+                            let node = builder.build(&mut self.node_classes.borrow_mut());
+
+                            Some((cached_node.id, node))
+                        })
+                        .unwrap() // Must be Some() because `is_dirty` returned true.
```

Comment:
```suggestion
                        ?
```
(and then you don't even need the  is_dirty() check before).